### DISCUSSION TITLE
How to disable the Names dictionary in code?

### DISCUSSION THREAD

**Discussion by konata-san** [melink14/rikaikun#2355] [Q&A]
Hi. I am trying to use the rikaikun extension in the Orion browser on my iPhone which used to work, but now is defaulting to the Names dictionary and I can't switch dictionaries since I can't use the keyboard shortcuts. Ideally it will only use one of the dictionaries that shows the normal word definitions.

I'm hoping I can compile a version of the extension without the Names dictionary and load it on my phone, but I had a quick look through the code and it's not super clear what change I would need to make in the code to do that.

Does anyone happen to know if there is a simple way I can disable the Names dictionary in code?

**Comment by konata-san** [comment]
That said, I have a feeling the issue might be worse than just defaulting to the Names dictionary since even if I connect a wireless keyboard to my phone I can use most of the keyboard shortcuts, but the shortcut for switching dictionaries doesn't work.

**Comment by konata-san** [comment]
I also tried using older versions of the extension from when it used to work and have the same problem, so I don't think it's an issue caused by an update to the extension but most likely caused by an update to the Orion browser.

**Comment by melink14** [comment]
I'm not sure why it would default to Names given that's the 3rd dictionary after regular words and Kanji...

What you could try is editing this method to ignore dictOption and always return ` e = this.dict.wordSearch(text, false);`

https://github.com/melink14/rikaikun/blob/7be51d911cd709c12e4a34e0b53404fadcbd337f/extension/rikaichan.ts#L273

Ideally, you could view some logs to see if any error messages were being posted but not sure how you would do that in an iOS browser unless you used an emulator or remote debugging.

**Comment by konata-san** [comment]
Thanks for the idea! Just gave that a go but unfortunately it just means no dictionary shows up.
I assume there is some bug that is preventing the first two dictionaries from loading...

Unfortunately there doesn't seem to be an easy way to debug this since I there doesn't seem to be a way to get to the console logs from the iOS browser.

Thanks for your help anyway.

The reason I was using rikaikun on my phone was to read books in Japanese on my iPhone with easy word lookups. Is there any chance you know of another good way to do that?

**Reply by melink14** [reply]
I'd give https://apps.apple.com/us/app/10ten-japanese-reader/id1573540634 a shot.

The author has done a lot of good work even though I don't use iOS myself.

**Reply by konata-san** [reply]
Thanks, I'll give that a go!
